config/overrides.go

```diff
@@ -361,3 +363,84 @@ func NamePrefixRemoval() tjconfig.ResourceOption {
 		r.ExternalName.OmittedFields = append(r.ExternalName.OmittedFields, "name_prefix")
 	}
 }
+
+// KnownReferencers adds referencers for fields that are known and common among
+// more than a few resources.
+func KnownReferencers() tjconfig.ResourceOption { //nolint:gocyclo
+	return func(r *tjconfig.Resource) {
+		for k, s := range r.TerraformResource.Schema {
+			if s.Computed && !s.Optional {
+				continue
+			}
+			switch {
+			case strings.HasSuffix(k, "role_arn"):
+				r.References[k] = tjconfig.Reference{
+					Type:      "github.com/crossplane-contrib/provider-jet-aws/apis/iam/v1alpha1.Role",
+					Extractor: common.PathARNExtractor,
+				}
+			case strings.HasSuffix(k, "security_group_ids"):
+				r.References[k] = tjconfig.Reference{
+					Type:              "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha1.SecurityGroup",
+					RefFieldName:      strings.TrimSuffix(name.NewFromSnake(k).Camel, "s") + "Refs",
```

Comment:
What about caching `strings.TrimSuffix(name.NewFromSnake(k).Camel, "s")` in a variable?

Reply:
I think it reads better to have it contained like this and there isn't much gain/loss in performance.